utils/coreutils/techutils.go

```diff
@@ -191,6 +204,183 @@ func DetectedTechnologiesList() (technologies []string) {
 	return techStringsList
 }
 
+// If recursive is true, the search will not be limited to files in the root path.
+// If requestedTechs is empty, all technologies will be checked.
+// If excludePathPattern is not empty, files/directories that match the wildcard pattern will be excluded from the search.
+func DetectTechnologiesDescriptors(path string, recursive bool, requestedTechs []string, requestedDescriptors map[Technology][]string, excludePathPattern string) (technologiesDetected map[Technology]map[string][]string) {
+	filesList, err := fspatterns.ListFiles(path, recursive, false, true, true, excludePathPattern)
+	if err != nil {
+		return
+	}
+	workingDirectoryToIndicators, excludedTechAtWorkingDir := mapFilesToRelevantWorkingDirectories(filesList, requestedDescriptors)
+	strJson, err := json.MarshalIndent(workingDirectoryToIndicators, "", "  ")
+	if err == nil && len(workingDirectoryToIndicators) > 0 {
+		log.Debug(fmt.Sprintf("mapped %d working directories with indicators/descriptors:\n%s", len(workingDirectoryToIndicators), strJson))
+	}
+	technologiesDetected = mapWorkingDirectoriesToTechnologies(workingDirectoryToIndicators, excludedTechAtWorkingDir, ToTechnologies(requestedTechs), requestedDescriptors)
+	if len(technologiesDetected) > 0 {
+		log.Debug(fmt.Sprintf("Detected %d technologies at %s: %s.", len(technologiesDetected), path, maps.Keys(technologiesDetected)))
+	}
+	return
+}
+
+// Map files to relevant working directories according to the technologies' indicators/descriptors and requested descriptors.
```

Comment:
Let's also describe the function and arguments and the returned values, including the maps structure.
The same applies for the `mapWorkingDirectoriesToTechnologies` function.